Util/RequestUtil.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+/*
+ * This file is part of the FOSRestBundle package.
+ *
+ * (c) FriendsOfSymfony <http://friendsofsymfony.github.com/>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace FOS\RestBundle\Util;
+
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * Utility class for request-related functions.
+ *
+ * @internal Should not be called by user code
+ *
+ * @author Florian Voutzinos <florian@voutzinos.com>
+ */
+final class RequestUtil
+{
+    /**
+     * Checks if the request is a form post.
+     *
+     * @param Request $request
+     *
+     * @return boolean
+     */
+    public static function isFormPost(Request $request)
+    {
+        if (!$request->isMethod('POST')) {
```

Comment:
If the request is PUT, it will enter the `if` and the normalization will happen (https://github.com/florianv/FOSRestBundle/blob/array-normalizer/EventListener/BodyListener.php#L119). 

When the feature was initially merged, we only normalized requests with a decodable format (non forms). Support for forms was added later through the `forms` option to keep BC.

Maybe this method should be private in the listener as @lsmith77 suggested because it's not really reusable indeed